holopy/scattering/tests/test_lens.py

```diff
@@ -155,64 +150,63 @@ def test_calc_holo_theta_npts_not_equal_phi_npts(self):
-    @unittest.skipUnless(NUMEXPR_INSTALLED, "numexpr package required")
-    def test_numexpr_integrand_prefactor1(self):
-        expr = LENSMIE.numexpr_integrand_prefactor1
-        # integrand_prefactor1 = 'exp(1j * krho_p * sinth * cos(phi - phi_p))'
-
-        krho_p, sinth, phi, phi_p = np.random.rand(4)
-
-        result_numpy = np.exp(1j * krho_p * sinth * np.cos(phi - phi_p))
-        result_numexpr = ne.evaluate(expr)
-        assert_equal(result_numpy, result_numexpr)
-
-    @unittest.skipUnless(NUMEXPR_INSTALLED, "numexpr package required")
-    def test_numexpr_integrand_prefactor2(self):
-        expr = LENSMIE.numexpr_integrand_prefactor2
-        # integrand_prefactor2 = 'exp(1j * kz_p * (1 - costh))'
-
-        kz_p, costh = np.random.rand(2)
-
-        result_numpy = np.exp(1j * kz_p * (1 - costh))
-        result_numexpr = ne.evaluate(expr)
-        assert_equal(result_numpy, result_numexpr)
-
-    @unittest.skipUnless(NUMEXPR_INSTALLED, "numexpr package required")
-    def test_numexpr_integrand_prefactor3(self):
-        expr = LENSMIE.numexpr_integrand_prefactor3
-        # integrand_prefactor3 = 'sqrt(costh) * sinth * dphi * dth'
-
-        costh, sinth, dphi, dth = np.random.rand(4)
-
-        result_numpy = np.sqrt(costh) * sinth * dphi * dth
-        result_numexpr = ne.evaluate(expr)
-        assert_equal(result_numpy, result_numexpr)
-
-    @unittest.skipUnless(NUMEXPR_INSTALLED, "numexpr package required")
-    def test_numexpr_integrandl(self):
-        expr = LENSMIE.numexpr_integrandl
-        # integrandl = ('prefactor * (cosphi * (cosphi * S2 + sinphi * S3) +'
-        #               + ' sinphi * (cosphi * S4 + sinphi * S1))')
-
-        prefactor, cosphi, sinphi, S1, S2, S3, S4 = np.random.rand(7)
-
-        result_numpy = prefactor * (cosphi * (cosphi * S2 + sinphi * S3)
-                                    + sinphi * (cosphi * S4 + sinphi * S1))
-        result_numexpr = ne.evaluate(expr)
-        assert_equal(result_numpy, result_numexpr)
-
-    @unittest.skipUnless(NUMEXPR_INSTALLED, "numexpr package required")
-    def test_numexpr_integrandr(self):
-        expr = LENSMIE.numexpr_integrandr
-        # integrandr = ('prefactor * (sinphi * (cosphi * S2 + sinphi * S3) -'
-        #                + ' cosphi * (cosphi * S4 + sinphi * S1))')
-
-        prefactor, cosphi, sinphi, S1, S2, S3, S4 = np.random.rand(7)
-
-        result_numpy = prefactor * (sinphi * (cosphi * S2 + sinphi * S3)
-                                    - cosphi * (cosphi * S4 + sinphi * S1))
-        result_numexpr = ne.evaluate(expr)
-        assert_equal(result_numpy, result_numexpr)
+    @unittest.skipUnless(lens.NUMEXPR_INSTALLED, "numexpr package required")
+    def test_integrand_prefactor_same_with_numexpr_as_without(self):
+        np.random.seed(1649)
+        krho, phi, kz = np.random.randn(3, 101)
+
+        prefactor_numexpr = LENSMIE._integrand_prefactor(krho, phi, kz)
+
+        # Monkey-patch lens to use the numpy pathway:
```

Comment:
I think this is a sign that some functionality is missing - e.g. calling Lens with a `computation_method='auto'` that can be overridden to specify numpy or numexpr. 

However, we also want to avoid complexifying [not a word] the interface, and there's no real reason you'd want to enforce numpy outside of testing so this seems fine.

Reply:
I notice you do this a few times through this test file. You could consider writing a context manager for the monkey patch. That would also make it less likely someone breaks things later since the numexpr adjustments would be removed from the tests themselves.

Reply:
Should be fixed by babd9bcd. Let me re-run the tests quickly on my computer with numexpr working before merging this in though.